src/main/java/org/dependencytrack/parser/cyclonedx/util/ModelConverter.java

```diff
@@ -348,6 +349,31 @@ public static org.cyclonedx.model.Metadata createMetadata(final Project project)
                 });
                 cycloneComponent.setExternalReferences(references);
             }
+            /*Issue #2737:  Adding supplier contact functionality */
+            if (project.getSupplier() != null) {
+                OrganizationalEntity supplier = new OrganizationalEntity();
+                supplier.setName(project.getSupplier().getName());
+            
+                if (project.getSupplier().getUrls() != null) {
+                    supplier.setUrls(project.getSupplier().getUrls());
+                } else {
+                    supplier.setUrls(null);
+                }
+                if (project.getSupplier().getContacts() != null) {
+                    List<OrganizationalContact> contacts = new ArrayList<>();
+                    for (org.cyclonedx.model.OrganizationalContact OrganizationalContact: project.getSupplier().getContacts()) {
```

Comment:
Probably a typo, but the loop variable should be of type `org.dependencytrack.model.OrganizationalContact`, as you're iterating over supplier contacts of the DT model.

```suggestion
                    for (OrganizationalContact organizationalContact: project.getSupplier().getContacts())) {
```

Reply:
made the change :-) will commit shortly.